api/v1alpha3/conversion.go

```diff
@@ -258,3 +298,36 @@ func restoreInstance(restored, dst *v1alpha4.Instance) {
 	}
 	dst.VolumeIDs = restored.VolumeIDs
 }
+
+func RestoreNonRootVolume(restoredVolumes, dstVolumes []v1alpha4.Volume) {
+	if restoredVolumes == nil {
+		return
+	}
+	if dstVolumes == nil {
+		dstVolumes = make([]v1alpha4.Volume, 0)
+	}
+	for i := range restoredVolumes {
+		if restoredVolumes[i].Encrypted == nil {
+			if len(dstVolumes) <= i {
+				dstVolumes = append(dstVolumes, restoredVolumes[i])
+			} else {
+				dstVolumes[i].Encrypted = nil
+			}
+		}
+	}
+}
+
+func RestoreRootVolume(restored, dst *v1alpha4.Volume) {
```

Comment:
Can we add some commentary in this as to why we have this logic so it's clear to other people looking at the code as to why we're doing this?